File: scripting/python/cdmsbats/ChargeChanRecord.py
# -*- coding: utf-8 -*-
"""
Created on Fri Apr 11 12:14:28 2014

@author: tdoughty1
"""

from DataRecord import DataRecord


class ChargeChanRecord(DataRecord):

    def __init__(self):

        self._detCode = None
        self._towerNum = None
        self._driverGain = None
        self._chanBias = None
        self._rtfOffset = None
        self._deltat = None
        self._t0 = None
        self._traceLength = None

    def StoreValues(self, Record):

        self._detCode = Record[0]
        self._towerNum = Record[1]
        self._driverGain = float(Record[2])/100
        self._chanBias = float(Record[3])/1e6
        self._rtfOffset = float(Record[4])/1e6
        self._deltat = float(Record[5])/1000
        self._t0 = float(Record[6])/1000
        self._traceLength = Record[7]

    def PrintValues(self):

        print "Detector Code = %d" % self._detCode
        print "Tower Number = %d" % self._towerNum
        print "Driver Gain = %3.1f" % self._driverGain
        print "Channel Bias = %5.3f V" % self._chanBias
        print "RTF Offset = %5.3f V" % self._rtfOffset
        print "Delta T = %4.2f us" % self._deltat
        print "t0 = %5.1f us" % self._t0
        print "Trace Length = %d" % self._traceLength

File: compiled/java/cdmsbats/src/batcommon/datareader/ChargeChanRecord.java
/**
 * 
 */
package batcommon.datareader;

/**
 * @author tdoughty1
 *
 */
public class ChargeChanRecord extends DataRecord {
	
	private int _detCode;
	private int _towerNum;
	private float _driverGain;
	private float _chanBias;
	private double _rtfOffset;
	private float _deltat;
	private float _t0;
	private int _traceLength;

	public ChargeChanRecord(){
		
		_detCode = -999999;
        _towerNum = -999999;
        _driverGain = -999999;
        _chanBias = -999999;
        _rtfOffset = -999999;
        _deltat = -999999;
        _t0 = -999999;
        _traceLength = -999999;
	}
	
	public void StoreValues(int[] Record){

        _detCode = Record[0];
        _towerNum = Record[1];
        _driverGain = (float)Record[2]/100;
        _chanBias = (float)Record[3]/100;
        _rtfOffset = (float)Record[6]/1e6;
        _deltat = (float)Record[8]/1000;
        _t0 = (float)Record[9]/1000;
        _traceLength = Record[10];
	}
	
	public void PrintValues(){
		
		System.out.printf("Detector Code = %d\n", _detCode);
        System.out.printf("Tower Number = %d\n", _towerNum);
        System.out.printf("Driver Gain = %3.1f\n", _driverGain);
        System.out.printf("Channel Bias = %6.2f pA\n", _chanBias);
        System.out.printf("RTF Offset = %4.2f V\n", _rtfOffset);
        System.out.printf("Delta T = %4.2f us\n", _deltat);
        System.out.printf("t0 = %5.1f us\n", _t0);
        System.out.printf("Trace Length = %d\n", _traceLength);
	}
}
